.github/workflows/docker-build.yaml

```diff
@@ -465,6 +477,73 @@ jobs:
+        if: |
+            always() && !failure() && !cancelled() &&
+            (
+                github.event.pull_request.head.repo.full_name == 'shopsys/shopsys' ||
+                github.ref_protected == true
+            ) &&
+            (needs.build-php-fpm-image.result == 'success' || needs.build-php-fpm-image.result == 'skipped') &&
+            (needs.build-storefront-image.result == 'success' || needs.build-storefront-image.result == 'skipped') &&
+            (needs.build-elasticsearch-image.result == 'success' || needs.build-elasticsearch-image.result == 'skipped')
+        runs-on: ubuntu-22.04
+        env:
+            GH_TOKEN: ${{ github.token }}
+        steps:
+            -   name: Check pull request labels
+                run: |
+                    PULL_REQUEST_LABELS=`gh api -H "Accept: application/vnd.github+json" /repos/shopsys/shopsys/issues/${{ github.event.pull_request.number }}/labels | jq '[.[].name]'`
+                    if ! echo "$PULL_REQUEST_LABELS" | jq -e '. | index("regenerate screenshots")' >/dev/null; then
+                        exit 1;
+                    fi
+            -   name: GIT checkout branch - ${{ github.head_ref || github.ref_name }}
+                uses: actions/checkout@v4
+                with:
+                    ref: ${{ github.head_ref || github.ref_name }}
+            -   name: Print github head ref
+                run: |
+                    echo ${{ github.head_ref }}
+            -   name: Print github ref name
+                run: |
+                    echo ${{ github.ref_name }}
+            -   name: Print github event.pull_request.head.ref
+                run: |
+                    echo ${{ github.event.pull_request.head.ref }}
+            -   name: Download docker-compose.cypress.yml from artifacts
+                uses: actions/download-artifact@v4
+                with:
+                    name: docker-compose-cypress
+            -   name: Build application
+                run: |
+                    mv docker-compose.cypress.yml docker-compose.yml
+                    docker compose up -d --wait
+                    docker compose exec php-fpm php phing -D production.confirm.action=y db-create frontend-api-generate-new-keys build-demo-dev-quick error-pages-generate elasticsearch-index-recreate elasticsearch-export
+            -   name: Run Cypress tests
+                run: |
+                    docker compose run -e TYPE=base -e COMMAND=run cypress-regenerate
+            -   name: Push snapshots to the branch
+                run: |
+                    cp -rf snapshots/* project-base/storefront/cypress/snapshots/
+                    git config --global user.name 'ShopsysBot'
+                    git config --global user.email 'shopsysbot@users.noreply.github.com'
+                    git add project-base/storefront/cypress/snapshots
+                    git commit -m "regenerate screenshots"
+                    git push
```

Comment:
what will happen when the label `regenerate screenshots` is present and you push another commit that does not change screenshots?

Reply:
The developer should remove the label before pushing new commits. If they don't, it will again trigger the regenerate workflow. If they notice that, they can "Cancel workflow", remove the label, then manually start all jobs again.

Reply:
I will include this use case in documentation and point it out in the upcoming educational meeting.